### PR TITLE
feat: add yargs-unparser libdefs

### PR DESCRIPTION
**Pull request by antongolub**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by antongolub** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/runs/2732162955?check_suite_focus=true
```
Error in yargs-unparser
Error: /home/runner/work/DefinitelyTyped/DefinitelyTyped/types/yargs-unparser/index.d.ts:1:1
ERROR: 1:1  npm-naming  The declaration doesn't match the JavaScript module 'yargs-unparser'. Reason:
The declaration should use 'export =' syntax because the JavaScript source uses 'module.exports =' syntax and 'module.exports' can be called or constructed.
```

Reasonable, but in this case **dtslint** fails with:
```
Error: /DefinitelyTyped/types/yargs-unparser/index.d.ts:8:19
ERROR: 8:19  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 26:1  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
ERROR: 27:1  strict-export-declare-modifiers  All declarations in this module are exported automatically. Prefer to explicitly write 'export' for clarity. If you have a good reason not to export this declaration, add 'export {}' to the module to shut off automatic exporting. See: https://github.com/Microsoft/dtslint/blob/master/docs/strict-export-declare-modifiers.md
```

So, `strict-export-declare-modifiers` should be disabled.

File: types/yargs-unparser/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Type definitions for yargs-unparser 2.0
+// Project: https://github.com/yargs/yargs-unparser#readme
+// Definitions by: Anton Golub <https://github.com/antongolub>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
```
Comment (peterblazejewicz) [inline types/yargs-unparser/index.d.ts:5]:
```suggestion
```
It's already unsupported on DT, 
let me know if you've used kind of template to create types (so TS 2.2 was added), maybe it needs an update, thx!

File: types/yargs-unparser/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+// tslint:disable:strict-export-declare-modifiers
```
Comment (peterblazejewicz) [inline types/yargs-unparser/index.d.ts:7]:
```suggestion
```
all your types are automatically exported, no need for this one IMO

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
+1 for Peter's comments and please address the mismatched form in tsconfig.json

**Comment by antongolub** [comment]
Fixed. I've run `tslint && dtslint` from `types/yargs-unparser`, instead of `npm test yargs-unparser`  from the pkg root. So I've tried to fix an issue that does not exist.

**Review by RyanCavanaugh — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by antongolub** [comment]
Ready to merge

**Pull request merged**